internal/manifests/manifests_test.go

```diff
@@ -198,20 +198,6 @@ var _ = Describe("ClusterManifestTests", func() {
 			Expect(responsePayload.Payload.Folder).To(Equal(defaultFolder))
 		})
 
-		It("cluster doesn't exist", func() {
```

Comment:
Why remove these tests? Shouldn't we still error in these cases?

Reply:
As I wrote in the PR description, the logic is not in the app layer anymore but in the middleware so those tests were moved to the subsystem test to make sure the system behaves as we expect when we depends on the authz handler

Reply:
we have but not here but in the subsystem. These function will not be called if the cluster does not exist

Reply:
> These function will not be called if the cluster does not exist

What about cases where we're running without auth or in kube-api?
The authz middleware doesn't check ownership in those scenarios.
We should have a sane response from these functions even if we're not using rhsso auth.

Reply:
From @avishayt in a slack conversation:

> If the question is "do we need to make sure that the cluster exists?", I would say "nice to have but no".  In SaaS, if someone deleted the cluster then the authz won't allow them to view the manifests.  In k8s API, manifests are read from ConfigMaps and not from the backend anyway (the controller lists manifests just to check which ones it already created in the backend).

I guess this means that whether the cluster exists in the database is actually irrelevant (before and after this change). I suppose we could have a test for when the manifest doesn't exist in s3, but that doesn't seem super important.